terraform/indexing-subsystem/force-trigger/bucket-trigger.js

```diff
@@ -0,0 +1,44 @@
+const { S3Client, ListObjectsV2Command } = require('@aws-sdk/client-s3')
+const { SQSClient, SendMessageCommand } = require('@aws-sdk/client-sqs')
+
+const S3client = new S3Client()
+const SQSclient = new SQSClient()
+
+async function* listAllKeys(opts) {
+  opts = { ...opts }
+  do {
+    const command = new ListObjectsV2Command(opts)
+    const foundObjects = await S3client.send(command)
+    opts.ContinuationToken = foundObjects.NextContinuationToken
+    yield foundObjects
```

Comment:
Not really used to generators but I think this code looks fine.

I would say just make sure the `yield` in the final iteration (not sure if it yields `[]`, `null` or `undefined`) doesn't make the `main` iteration fail.